src/librustc_typeck/check/_match.rs

```diff
@@ -50,30 +50,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         self.warn_arms_when_scrutinee_diverges(arms, match_src);
 
-        // Otherwise, we have to union together the types that the
-        // arms produce and so forth.
-        let scrut_diverges = self.diverges.get();
-        self.diverges.set(Diverges::Maybe);
+        // Otherwise, we have to union together the types that the arms produce and so forth.
+        let scrut_diverges = self.diverges.replace(Diverges::Maybe);
```

Comment:
Ohh, we might have more similar places where we use `get` and `set`, predating `Cell::replace`.
I found some in these two files, if someone wants to take them:
* `src/librustc/ty/print/pretty.rs`
* `src/librustc/infer/mod.rs`

(there's others, but they use the result of `get()` to compute the value to `set`, and I don't think we have a nice abstraction for that)